src/csvcubed/utils/sparql_handler/data_cube_inspector.py

```diff
@@ -207,6 +217,50 @@ def get_code_lists_and_cols(self, csv_url: str) -> CodelistsResult:
 
         return self._codelists_and_cols.get(csv_url, CodelistsResult([], 0))
 
+    def get_dataframe(self, csv_url: str) -> Tuple[pd.DataFrame, List[ValidationError]]:
+        """
+        Get the pandas dataframe for the csv url of the cube wishing to be loaded.
+        """
+        cols = self.get_column_component_info(csv_url)
+        dict_of_types = {}
+        for col in cols:
```

Comment:
This function seems to be getting a bit big now. Could you simplify it a bit by refactoring some of the code into a separate function? Perhaps the stuff which generates the `dict_of_types` makes sense to put into a separate function?

Reply:
refactored the `dict_of_types` bit into a private function.